W-16161756 download from google drive (#411)
mulesoft/docs-rpa: rpa-builder/6.7/modules/ROOT/pages/toolbox-google-workspace-download-from-google-drive.adoc

= Download from Google Drive 

> Download or Export files from your Google Drive. 

Reviewer: Summarize topic & give user value in the s/d - particularly at the H1 level - for SEO/AI.

> You can export Google Docs, Sheets, and Slides into a selection of supported formats including PDF.

Reviewer: ```suggestion
You can export Google Docs, Sheets, and Slides to several supported formats, including PDF.
```

== Before You Begin 

> Before downloading a file, ensure you: 

Reviewer: You could make these present tense/imperative.

Before downloading a file:
- Configure
- Obtain

* Configured your Google API Service Account Credentials.
+

> You can configure your credentials as an xref:toolbox-variable-handling-activity-parameters.adoc#defining-activity-parameters[Activity Parameter] or inline in the action step wizard. 

Reviewer: Should this be "Action Step wizard"?

* Obtained the corresponding File ID by using the Search in Google Drive action step. 

> == Properties 

Reviewer: See prev comments about these reference sections. 

Also, it just occurred to me that it would be better to put these after the task. Start with the actual task and then link to the reference material from that task, as necessary. We want to avoid making people read through reference material before they can get to the task.

Reviewer: (globally)

Reviewer: (If all sentence fragments, no period)

Reviewer: I think a table would be easier to scan.

Author: I agree with rearranging the topic and making properties into a table for better reading. However, this current structure is replicated in all RPA Builder action steps, so changing this one would make it inconsistent with the rest of the existing topics. This change should be made to all topics which, at the moment, is not a possibility.

* *Authentication Settings* 
** *Auth Provider x509 Cert URL*
+
Authentication provider certificate URL for the Google API credentials. 
** *Auth URI* 
+
OAuth2 authentication URI for the Google API credentials.
** *Client Email* 
+
Client email of the Google API credentials. 
** *Client ID* 
+
Client ID of the Google API credentials. 
** *Client x509 Cert URL* 
+
Client certificate URL for the Google API credentials. 
** *Credential Type*
+

> Type of Google API credentials. For example, `service_account`.

Reviewer: ```suggestion
Type of Google API credentials, for example, `service_account`.
```

** *Google API Service Account Credentials* 
+
Activity parameter containing all Google API credentials. 
** *Project ID* 
+
Project ID of the Google API credentials.
** *Token URI* 
+
OAuth2 token URI for the Google API credentials.
** *Universe Domain* 
+
Universe domain of the Google API credentials.

* *Common* 
+
See xref:common-properties.adoc[Common Properties]
+

> ** *Docs type* 

Reviewer 2: I think the whole Google Docs type section should be moved behind the Operation mode, so that Google Docs, Sheets and Slides with their export formats are after each other

Reviewer 2: Move line 54 - 64 to after line 79

Author: I copied the properties as they are listed in the Workflow Properties panel. I reordered them here in the docs as you suggested because it makes more sense.

+
ID of the file to download or export.
** *Google drive file type*
+

> Type of Google file to export. Available types are: 

Reviewer: Be consistent with line 70, 82, 93?

+
*** *Google Docs*
*** *Google Sheets*
*** *Google Slides*
+
** *Operation mode*
+
Operation to perform, either *DownloadFile* or *ExportDocument*.
** *Docs type* 
+
File format for exporting Google Docs. Available types: 
+

> *** *Microsoft Word (.docx)*

Reviewer: Is the extension part of the UI element?

Author: Yes, all properties have the same name as they show in the UI.

*** *OpenDocument (.odt)*
*** *Rich Text (.rtf)*
*** *PDF (.pdf)*
*** *Plain Text (.txt)*
*** *Web Page (HTML) (.zip)*
*** *EPUB (.epub)*
+
** *Sheets Type*
+
File format for exporting Google Sheets. Available types:
+
*** *Microsoft Excel (.xlsx)*
*** *OpenDocument (.ods)*
*** *PDF (.pdf)*
*** *Web Page (HTML) (.zip)*
*** *Comma-Separated Values (first-sheet only) (.csv)*
*** *Tab-Separated Values (first-sheet only) (.tsv)*
+
** *Slides Type*
+
File format for exporting Google Slides. Available types:
+
*** *Microsoft PowerPoint (.pptx)*
*** *ODP (.odp)* 
*** *PDF (.pdf)* 
*** *Plain Text (.txt)*
*** *JPEG (first-slide only) (.jpg)*
*** *PNG (first-slide only) (.png)*
*** *Scalable Vector Graphics (first-slide only) (.svg)*

* *Google Drive File Settings*
** *Create directory if it does not exist*
+

> When selected, creates the folder specified in *Directory path* if it does not exist.

Reviewer: ```suggestion
If selected, creates the folder specified in *Directory path* if it doesn’t exist.
```

** *Directory path* 
+

> Path to the directory where to download or export the file. 

Reviewer: ```suggestion
Path to the directory to download or export the file. 
```

** *New file name* 
+

> Filename to use when *Set new file name* is enabled. You must specify the file name including the extension. 

Reviewer: ```suggestion
File name to use if *Set new file name* is enabled. You must specify the file name, including the extension. 
```

Reviewer: I believe our style is two words.

** *Overwrite existing file* 
+
When selected, overwrites any existing file in the specified download directory that has the same name as the downloaded or exported file.
** *Set new file name*
+
When selected, renames the downloaded or exported file using the specified *New file name*.

== Inbound Variables 

* *File ID*
* *Directory Path* 
* *Create directory if it does not exist*
* *Overwrite existing file* 
* *Set new file name* 
* *New file name* 
* *Credential Type*
* *Project ID* 
* *Private Key ID* 
* *Private Key* 
* *Token URI* 
* *Auth Provider x509 Cert URL* 
* *Auth URI* 
* *Client Email* 
* *Client ID* 
* *Client x509 Cert URL* 
* *Universe Domain* 
* *Google API Service Account Credentials* 

== Outbound Variables 

* *Download Succeeded* (Boolean)
+

> Confirms if the download was successful. 

Reviewer: Should this be parallel with the others? 

Boolean indicating whether the download was successful (true).

* *Target folder* 
+
String containing the directory path to the downloaded or exported file. 
* *File name* 
+
String containing the file name, including the extension. 

> == Wizard 

Reviewer: I really think this should be a task (imperative) heading describing what the user wants to do. The problem with these single-word generic headings is that search won't catch what the user is looking for - and might return multiple "Wizard" headings.

Download or Export a Google Workspace Document (or something like that)

Author: I tried my best to convert these to tasks because users don't actually perform a task here, they configure the action step to perform the tasks when the automation runs.

> Specify your authentication settings for the Google API service account in the *Authentication Settings* panel. Select *Pin credentials* to specify the activity parameter containing the credentials, or select *Use custom settings* to provide the information inline. You can either import a JSON file with your Google API credentials information or specify the authentication information manually. 

Reviewer: Can you make this a numbered list?

In *Download Settings*, select between *Download file* or *Export Google Workspace document*, depending on the operation to perform. When exporting a file, you must select the type of Google file to export (Docs, Sheets, or Slides) and the format for the target file. Available export formats depend on the file being exported. 

> Provide the *File ID* of the file to download or export. To obtain this ID, use the Search in Google Drive action step. 

Reviewer: Our style says not to use UI elements in this way. Suggest lowercase "file ID" and no bold.

Should the action step be init cap and bold? I think Simone does this. See this [example](https://github.com/mulesoft/docs-rpa/pull/296/files#:~:text=You%20use%20the%20*Search%20Pattern*%20Action%20Step%20to%20search%20for%20this%20arrow%20up%20symbol%20and%20limit%20the%20search%20area%20to%20the%20space%20directly%20below%20the%20column%20name.).

> Specify the *Directory path* where you want to download or export the file from your Google Drive. You can specify this action step to create the target directory if it doesn't exist and to overwrite any existing file in the directory path that contains the same name as the file to download or export. 

Reviewer: Same comment as 159

+
Path to the directory where to download or export the file. 
** *New file name* 
+
Filename to use when *Set new file name* is enabled. You must specify the file name including the extension. 
** *Overwrite existing file* 
+
When selected, overwrites any existing file in the specified download directory that has the same name as the downloaded or exported file.
** *Set new file name*
+
When selected, renames the downloaded or exported file using the specified *New file name*.

== Inbound Variables 

* *File ID*
* *Directory Path* 
* *Create directory if it does not exist*
* *Overwrite existing file* 
* *Set new file name* 
* *New file name* 
* *Credential Type*
* *Project ID* 
* *Private Key ID* 
* *Private Key* 
* *Token URI* 
* *Auth Provider x509 Cert URL* 
* *Auth URI* 
* *Client Email* 
* *Client ID* 
* *Client x509 Cert URL* 
* *Universe Domain* 
* *Google API Service Account Credentials* 

== Outbound Variables 

* *Download Succeeded* (Boolean)
+
Confirms if the download was successful. 
* *Target folder* 
+
String containing the directory path to the downloaded or exported file. 
* *File name* 
+
String containing the file name, including the extension. 

> * *Success* 

Author: ```suggestion
```
